pkgs/servers/caddy/default.nix

```
@@ -2,35 +2,23 @@
 
 buildGoModule rec {
   pname = "caddy";
-  version = "1.0.5";
+  version = "2.0.0";
 
-  subPackages = [ "caddy" ];
+  subPackages = [ "cmd/caddy" ];
 
   src = fetchFromGitHub {
     owner = "caddyserver";
     repo = pname;
     rev = "v${version}";
-    sha256 = "0jrhwmr6gggppskg5h450wybzkv17iq69dgw36hd1dp56q002i7g";
+    sha256 = "1c1frfx0qkprhf4var70cncvrw8s9gjag2hygndbd9055hb52bvv";
   };
-  vendorSha256 = "09vnci9pp8zp7bvn8zj68wslz2nc54nhcd0ll31sqfjbp00215mj";
-
-  doCheck = false;
 
-  preBuild = ''
-    cat << EOF > caddy/main.go
-    package main
-    import "github.com/caddyserver/caddy/caddy/caddymain"
-    func main() {
-      caddymain.EnableTelemetry = false
-      caddymain.Run()
-    }
-    EOF
-  '';
+  vendorSha256 = "09vnci9pp8zp7bvn8zj68wslz2nc54nhcd0ll31sqfjbp00215mj";
```

This checksum is incorrect. This is an output derivation, so if you already had Caddy built with `09vnci9pp8zp7bvn8zj68wslz2nc54nhcd0ll31sqfjbp00215mj` vendor it would still continue to work on your computer despite incorrect checksum.

```suggestion
  vendorSha256 = "004hpjxpp18f71vy5v5ky0g07a8d5xh5qwl5b4bbx34hpf8yxs81";
```

---

You're right, it should now be fixed. The tests are still failing for the same unknown reasons, but 2.1.1 now builds successfully.